clustering/preprocess.py

```diff
@@ -19,8 +19,28 @@ def filter_restaurants(businesses, max_restaurants=None):
     (determined by looking at the review data).
     """
 
-    # TODO: filter the input DataFrame
-    return businesses
+    cols = ['business_id', 'is_open', 'review_count', 'categories']
+    businesses = businesses[cols]
+
+    businesses = businesses[businesses['review_count'] > 10]
+    businesses = businesses[businesses['is_open'] != 0]
+
+    valid_b = businesses.as_matrix()
+    b = np.empty_like(businesses['business_id'])
+
+    for i in range(businesses.shape[0]):
+        if "Food" or "Restaurant" or "food" or "restaurant" in valid_b[i, 3]:
```

Comment:
Does this comparison actually work? AFAIK the `or` comparisons take precedence, so the behaviour isn't what you might expect (unless something here is being overloaded somehow):

```python
>>> "a" or "b" or "c" or "d"
'a'
>>> bool("a" or "b" or "c" or "d" in ["e"])
True
```